avro/src/de.rs

```diff
@@ -230,6 +241,60 @@ impl<'de> de::VariantAccess<'de> for EnumDeserializer<'de> {
     }
 }
 
+impl<'de> de::EnumAccess<'de> for UnionDeserializer<'de> {
+    type Error = Error;
+    type Variant = Self;
```

Comment:
I am not sure this type is helpful here.
It usage below (Self::Variant) makes it more confusing.
Can we use just `Self` ?!